app/crud/scorecardServiceMetric.py

```diff
@@ -0,0 +1,122 @@
+from sqlalchemy.orm import Session
+from app.schemas.scorecardServiceMetric import ScorecardServiceMetricCreate, ScorecardServiceMetric, ScorecardServiceMetricUpdate
+from . import CRUDScoreCardMetric, CRUDMicroserviceScoreCard, CRUDMetric, CRUDScoreCard, CRUDMicroservice
+from .base import CRUDBase
+from app.utils import format_code
+from app.utils.base import parse_stringified_value
+
+
+class CRUDScoreCardServiceMetric(CRUDBase[ScorecardServiceMetric, ScorecardServiceMetricCreate, ScorecardServiceMetricUpdate]):
+    def __init__(self, db_session: Session):
+        self.scoreCard = CRUDScoreCard(db_session)
+        self.microServiceScoreCard = CRUDMicroserviceScoreCard(db_session)
+        self.scoreCardMetric = CRUDScoreCardMetric(db_session)
+        self.microService = CRUDMicroservice(db_session)
+        self.metric = CRUDMetric(db_session)
+
+    def getwithscorecardIDmetricandservice(self, scorecardID: int):
+        # it handles if the scorecard is not found & raises error message
+        scorecard = self.scoreCard.get(scorecardID)
+
+        # get ids of metrics and services used in this scorecard
+        metrics = self.scoreCardMetric.getbyscorecardID(scorecardID)
+        services = self.microServiceScoreCard.getservice(scorecardID)
+        metric_ids = [sc_id.metricId for sc_id in metrics]
+        service_ids = [sc_id.microserviceId for sc_id in services]
+
+        metricOBJs = []
+        serviceOBJs = []
+
+        metricCreatedList = self.metric.getByIds(metric_ids)
+        metricList = self.scoreCardMetric.getByMetricIdsandScorecardId(
+            metric_ids, scorecardID)
+        for metricCreated in metricCreatedList:
+            for metric in metricList:
```

Comment:
you don't need  the nested loops here, what you actually want is to map each metric_id from the metricList to its metric type.

```
metrics = self.metric.getByIds(metric_ids)
metrics_types_map = {metric.id : metric.type for metric in metrics}
for metric in metricList:
     metricType = metrics_types_map[metric.metricId]
    .....
```